frontend/src/hooks/useDeepCompareEffect.ts

```diff
@@ -0,0 +1,18 @@
+import { useRef, useEffect, EffectCallback, DependencyList } from 'react';
+import deepEqual from 'dequal';
+
+const useDeepCompareMemoize = (value: any) => {
+  const ref = useRef();
+
+  if (!deepEqual(value, ref.current)) {
+    ref.current = value;
+  }
+
+  return ref.current;
+};
+
+const useDeepCompareEffect = (callback: EffectCallback, dependencies?: DependencyList) => {
+  useEffect(callback, useDeepCompareMemoize(dependencies));
+};
+
+export default useDeepCompareEffect;
```

Comment:
Use this hook if you want to avoid having to use `React.memo` on some/every of the hook inputs. Found it handy